### PR TITLE
Search Functionality

### PR DESCRIPTION
**Pull request by andy-mtng**
The following pull request implements a search feature to view other player profiles.

### CODE REVIEW

**Review by jmmabanta — DISMISSED** [review]

**Comment by jmmabanta** [comment]
@WillQizza double check if it doesn't conflict with your leaderboard work (ui-wise)

File: app/src/main/java/com/cmput301w23t09/qrhunter/leaderboard/SearchQueryEntryAdapter.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+   * @param position Position of the search entry
+   * @param convertView Potential view the adapter can reuse
+   * @param parent Parent of convertView
+   * @return View that represents the search query entry
+   */
+  @Override
+  public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
+    View view = convertView;
+    if (view == null) {
+      view = LayoutInflater.from(context).inflate(R.layout.search_query_entry_view, parent, false);
```
Comment (WillQizza) [inline app/src/main/java/com/cmput301w23t09/qrhunter/leaderboard/SearchQueryEntryAdapter.java:50]: Seeing as we've now merged avatars, we should probably display them on this fragment too.

**Review by WillQizza — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/cmput301w23t09/qrhunter/leaderboard/LeaderboardFragment.java [hunk 2]
```diff
@@ -187,4 +193,22 @@ private void renderLeaderboards(List<Leaderboard<?>> leaderboards) {
...
     }
   }
+
+  /** Sets up the search view to respond to user input */
+  private void setUpPlayerSearch() {
+    playerSearchView.setOnQueryTextListener(
+        new SearchView.OnQueryTextListener() {
+          @Override
+          public boolean onQueryTextSubmit(String query) {
+            Toast.makeText(getContext(), query, Toast.LENGTH_SHORT).show();
```
Comment (WillQizza) [inline app/src/main/java/com/cmput301w23t09/qrhunter/leaderboard/LeaderboardFragment.java:203]: any particular reason for this?

**Review by WillQizza — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/cmput301w23t09/qrhunter/leaderboard/LeaderboardFragment.java (hunk 2 above)
Comment (andy-mtng) [reply app/src/main/java/com/cmput301w23t09/qrhunter/leaderboard/LeaderboardFragment.java:203]: That was originally just meant for debugging purposes but I forgot to remove it. Thanks for pointing it out William!

**Review by WillQizza — APPROVED** [review]

**Review by WillQizza — APPROVED** [review]

**Comment by andy-mtng** [comment]
I was looking over the code again and I just needed to make some very minor java doc changes. Should I wait for the github actions to complete before adding it?

**Comment by WillQizza** [comment]
> I was looking over the code again and I just needed to make some very minor java doc changes. Should I wait for the github actions to complete before adding it?
feel free to apply them

**Review by IreneRSun — APPROVED** [review]

**Pull request merged**